src/main/java/org/reactome/release/dataexport/FTPFileUploader.java

```diff
@@ -407,22 +407,26 @@ private void throwIllegalStateExceptionUnlessRequiredPropsPresent() {
 		}
 	}
 
-	private boolean uploadFileToServer(String fileToUpload) throws IOException {
+	boolean uploadFileToServer(String fileToUpload) throws IOException {
 		logger.info("Uploading file '{}' to server {}", fileToUpload, getServerHostName());
 
 		InputStream fileToUploadInputStream = new FileInputStream(fileToUpload);
 
+		boolean isUploadSuccessful;
 		if (getFtpClientToServer().storeFile(fileToUpload, fileToUploadInputStream)) {
 			logger.info("Successfully uploaded '{}' to server {}", fileToUpload, getServerHostName());
-			return true;
+			isUploadSuccessful = true;
 		} else {
 			logger.error("Unable to upload '{}' to the server {}.  FTP client reply message was: {}",
 				fileToUpload, getServerHostName(), getFtpClientReplyMessage());
-			return false;
+			isUploadSuccessful = false;
 		}
+		fileToUploadInputStream.close();
```

Comment:
Could a try-with-resources block be used to automatically close `fileToUploadInputStream`

Reply:
Also, does this fix the issue with passive FTP?

Reply:
I'm not sure if it was actually an issue with passive FTP.  I created IT tests for both the EuropePMC and NCBI file uploader and they create, upload, and delete empty dummy files to both servers successfully.

Reply:
So was the failure because the program did not close the `FileInputStream`? Using a try-with-resources block ensures than you don't need to call `.close()` explicitly, and can help avoid situations such as this one. ;)

Reply:
I'm not sure.  I can't run the entire program with the FTP upload without affecting the contents on the real servers.  I added the IT tests to upload and remove dummy files to the real servers to test that access and writing was okay, instead.  We can keep an eye on this next release to make sure the upload works properly.

Fair point - I will try to get in the habit of using try-with-resources :).